<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>647</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>std::vector demo</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_18">
    <item>
     <widget class="QListWidget" name="list_widget"/>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLineEdit" name="txt_elem_content"/>
        </item>
        <item>
         <widget class="QPushButton" name="B_edit">
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Edit</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_find">
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>find</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLineEdit" name="le_count">
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_count">
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>count</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lbl_count">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Методы</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QPushButton" name="B_pop_back">
          <property name="text">
           <string>pop_back</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_push_back">
          <property name="text">
           <string>push_back</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_15">
        <item>
         <widget class="QPushButton" name="B_pop_front">
          <property name="text">
           <string>pop_front</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_push_front">
          <property name="text">
           <string>push_front</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QPushButton" name="B_erase">
          <property name="text">
           <string>erase</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_insert">
          <property name="text">
           <string>insert</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QPushButton" name="B_clear">
          <property name="text">
           <string>clear</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Алгоритмы</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QPushButton" name="B_min_element">
          <property name="text">
           <string>min_element</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_max_element">
          <property name="text">
           <string>max_element</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_12">
        <item>
         <widget class="QPushButton" name="B_sort">
          <property name="text">
           <string>merge sort</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_sOrT">
          <property name="text">
           <string>merge sOrT</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_13">
        <item>
         <widget class="QPushButton" name="B_unique">
          <property name="text">
           <string>unique</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_reverse">
          <property name="text">
           <string>reverse</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_14">
        <item>
         <widget class="QPushButton" name="B_shuffle">
          <property name="text">
           <string>shuffle</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_16">
        <item>
         <widget class="QPushButton" name="B_lower_bound">
          <property name="text">
           <string>lower_bound</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_upper_bound">
          <property name="text">
           <string>upper_bound</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Итераторы</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QPushButton" name="B_minus">
          <property name="text">
           <string>--</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_plus">
          <property name="text">
           <string>++</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QPushButton" name="B_begin">
          <property name="text">
           <string>begin</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_end">
          <property name="text">
           <string>end</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Заготовки</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QPushButton" name="B_tea">
          <property name="text">
           <string>= чай</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="B_cake">
          <property name="text">
           <string>= торты</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout"/>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>size():</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="txt_size"/>
        </item>
        <item>
         <widget class="QPushButton" name="B_resize">
          <property name="text">
           <string>resize</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
